src/getOpenAIBody.ts

```diff
@@ -15,26 +35,15 @@ export function getOpenAIBody(
   const completionArgs = completionConfig.state.args
 
   const template = parsedBody.template ?? completionConfig.state.template
-  const inputMessages = [
-    ...template.map((item) => {
-      const needsCompilation =
-        typeof item.content === "string" ? item.content?.includes("{{") : true
-
-      const variables = Object.assign(
-        completionConfig.chatInput,
-        parsedBody.variables ?? {},
-      )
-      return {
-        ...item,
-        content:
-          item.content &&
-          (needsCompilation
-            ? compileLTTemplate(item.content, variables)
-            : item.content),
-      }
-    }),
-    ...(parsedBody.messages ?? []),
+  const variables = Object.assign(
+    completionConfig.chatInput,
+    parsedBody.variables ?? {},
+  )
+  const inputMessages = completionConfig.threadMessages ? compileMessages(completionConfig.threadMessages, variables) : [
```

Comment:
`completionConfig.threadMessages` should already be raw messages with no handlebars - no need to call compileMessages on these